### DISCUSSION TITLE
established a connection works only with ipsec restart

### DISCUSSION THREAD

**Discussion by tamirshkolnik** [strongswan/strongswan#1883] [General]
I'm trying to establish a connection between 2 Ubuntu machines. Ubuntu 22.04.2 LTS
Both machines running Strongswan 5.9.5
I update ipsec.conf and ipsec.secrets in both machines, than execute the command:
ipsec reload
ipsec update

But the connection failed to be established with the following error:
```
initiating IKE_SA tunnel-vpc1-to-2[10] to 10.11.4.4
generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
sending packet: from 10.11.4.3[500] to 10.11.4.4[500] (1044 bytes)
received packet: from 10.11.4.4[500] to 10.11.4.3[500] (344 bytes)
parsed IKE_SA_INIT response 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(CHDLESS_SUP) N(MULT_AUTH) ]
selected proposal: IKE:AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_1024
authentication of '10.11.4.3' (myself) with pre-shared key
no shared key found for '10.11.4.3' - '10.11.4.4'
```

If I execute ipsec restart on both sides the connection is working as expected
Also tried to use ipsec down/up with no success

Of course, I want to apply the new connection without restarting and disconnecting other open connections

what do I miss?

**Comment by tobiasbrunner** [comment]
> than execute the command:
> ipsec reload
> ipsec update

Only use the latter, the former first deletes all configs (even unmodified ones). Neither of them (re-)loads ipsec.secrets or the ipsec.d directory. For that, the respective [`ipsec reread*` commands](https://wiki.strongswan.org/projects/strongswan/wiki/ipseccommand#Reread-Commands) can be used (for secrets e.g. `ipsec rereadsecrets`).

Also, please consider using swanctl.conf.

**Reply by tamirshkolnik** [reply]
thanks for the quick answer.
your advice works great for me and now I trying to use the swanctl.
Considering I'm using swanctl, what is the corresponding command for reread/rereadsecrets in swanctl? because I'm editing the connections file (currently swanctl.conf) and want to refresh the connections without hangup the existing ones

**Reply by tobiasbrunner** [reply]
That's the default for swanctl/vici (depending on `start_action` and whether anything changed).

**Reply by tamirshkolnik** [reply]
ok
but still cannot establishes a connection using swanctl.
here is my swanctlc.conf:
```
connections {
	tunnel-vpc1-to-2 {
		include /etc/ipsec.d/ecp/swanctl/conf.d/ike_sa_default.conf
		remote_addrs=10.11.4.4
		local_addrs=10.11.4.3
		children {
			 tunnel-vpc1-to-2 {
			 	include /etc/ipsec.d/ecp/swanctl/conf.d/child_sa_default.conf
			 	start_action=start
			 	local_ts=10.0.0.0/27
			 	remote_ts=11.0.0.0/27
			 }

		}
		local-0 {
			id = 10.11.4.3
		}
		remote-0 {
			id = 10.11.4.4
		}
	}
}
pools {
}
authorities {
}
secrets {
	ike-10-11-4-310-11-4-4 {
		secret = "blabla"
		id-0=10.11.4.3
		id-2=10.11.4.4
	}
}
```

And this is the /etc/ipsec.d/ecp/swanctl/conf.d/ike_sa_default.conf:
```
# Global defaults for IKE_SA settings; Applied using include mechanism
rekey_time=28800s
version=2
proposals=aes256-sha256-modp1024
dpd_timeout=120s
dpd_delay=30s
```

But I'm getting the following error:

[IKE] no private key found for '10.11.4.4'

**Reply by misch42** [reply]
Please NEVER EVER publish PSKs on github. Please also change the key given above.

**Reply by misch42** [reply]
Please try:
secrets {
  ike-1 {
    secret = "blabla"
    id =10.11.4.3
  }
  ike-2 {
    secret = "blabla"
    id =10.11.4.4
  }
}

Also see: https://www.strongswan.org/testing/testresults/ikev2/rw-psk-ipv4/moon.swanctl.conf

**Reply by tobiasbrunner** [reply]
If you want to use PSKs, configure that as authentication method in local/remote sections (`auth=psk`, the default is `pubkey`, which is why you see a failed search for a private key).
